### PR TITLE
Add missing header

### PR DESCRIPTION
**Pull request by swills**

### CODE REVIEW

**Comment by wwmm** [comment]
Hi! I do not see a problem in adding this include but I am curious about the compiler error. As the source is right now it compiles in different linux distributions with no problem. So maybe this is not the real solution to the problem you are facing.

As far as I remember `utils.cpp` dos not use what is in the `array` include. The error came from somewhere else that includes `utils.hpp`. Again the source builds fine here so I don't think there is a problem in the source

**Comment by wwmm** [comment]
Adding the include to the file `plugin_ui_base.hpp` would be more appropriate. Which compiler version are you using? In which system? At least on Arch Linux with gcc `8.2.1` I can't reproduce this compilation problem

**Comment by wwmm** [comment]
There is no need to do a pull request. I will add it before releasing version `4.4.1`

Considering I am able to build with `clang 7` in Arch Linux this may be a clang 6 bug or something specific to freebsd. In any case there is no harm in adding the include. I just wanted to have a better idea of the cause

**Comment by swills** [comment]
Perhaps it's related to libc++ which is used instead of libstdc++?

**Comment by wwmm** [comment]
I don't know. I have no idea about the differences between them.

In any case your installation has the header. But for some reason it has to be explicitly included. Something must have changed in the newer compilers and there is no need to do this anymore.

**Comment by swills** [comment]
I'll try building with clang 7 and let you know.

**Comment by swills** [comment]
Yeah, same issue with clang 7:

```
FAILED: src/src@@pulseeffects@exe/sink_input_effects_ui.cpp.o                                                                                                                                  
clang++70 -Isrc/src@@pulseeffects@exe -Isrc -I../src -I../include -I. -I../ -Idata -I/usr/local/include -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include/gl
ibmm-2.4 -I/usr/local/lib/glibmm-2.4/include -I/usr/local/include/sigc++-2.0 -I/usr/local/lib/sigc++-2.0/include -I/usr/local/include/gtk-3.0 -I/usr/local/include/pango-1.0 -I/usr/local/inclu
de/fribidi -I/usr/local/include/cairo -I/usr/local/include/pixman-1 -I/usr/local/include/freetype2 -I/usr/local/include/libdrm -I/usr/local/include/libpng16 -I/usr/local/include/harfbuzz -I/u
sr/local/include/gdk-pixbuf-2.0 -I/usr/local/include/gio-unix-2.0/ -I/usr/local/include/atk-1.0 -I/usr/local/include/at-spi2-atk/2.0 -I/usr/local/include/at-spi-2.0 -I/usr/local/include/dbus-
1.0 -I/usr/local/lib/dbus-1.0/include -I/usr/local/include/gtkmm-3.0 -I/usr/local/lib/gtkmm-3.0/include -I/usr/local/include/atkmm-1.6 -I/usr/local/include/giomm-2.4 -I/usr/local/lib/giomm-2.
4/include -I/usr/local/include/pangomm-1.4 -I/usr/local/lib/pangomm-1.4/include -I/usr/local/include/cairomm-1.0 -I/usr/local/lib/cairomm-1.0/include -I/usr/local/include/gtk-3.0/unix-print -
I/usr/local/include/gdkmm-3.0 -I/usr/local/lib/gdkmm-3.0/include -I/usr/local/include/gstreamer-1.0 -I/usr/local/include/lilv-0 -I/usr/local/include/serd-0 -I/usr/local/include/sord-0 -I/usr/
local/include/sratom-0 -I/usr/local/include/bs2b -Xclang -fcolor-diagnostics -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++17 -O3 '-DG_LOG_DOMAIN="pulseeffects"'
 -O2 -fstack-protector -isystem /usr/local/include -fno-strict-aliasing -isystem /usr/local/include -isystem /usr/local/include -pthread -D_THREAD_SAFE -D_REENTRANT  -MD -MQ 'src/src@@pulseef
fects@exe/sink_input_effects_ui.cpp.o' -MF 'src/src@@pulseeffects@exe/sink_input_effects_ui.cpp.o.d' -o 'src/src@@pulseeffects@exe/sink_input_effects_ui.cpp.o' -c ../src/sink_input_effects_ui
.cpp                                                                                                                                                                                           
In file included from ../src/sink_input_effects_ui.cpp:1:                                                                                                                                      In file included from ../include/sink_input_effects_ui.hpp:4:                                                                                                                                  In file included from ../include/autogain_ui.hpp:5:                                                                                                                                            ../include/plugin_ui_base.hpp:61:21: error: type 'const std::array<double, 2>' does not provide a subscript operator                                                                               auto left = peak[0];                                                                                                                                                                                       ~~~~^~                                                                                                                                                                         ../include/plugin_ui_base.hpp:62:22: error: type 'const std::array<double, 2>' does not provide a subscript operator                                                                               auto right = peak[1];                                                                                                                                                                                       ~~~~^~                                                                                                                                                                        ../include/plugin_ui_base.hpp:89:21: error: type 'const std::array<double, 2>' does not provide a subscript operator                                                                               auto left = peak[0];                                                                                                                                                                       
                ~~~~^~                                                                                                                                                                         ../include/plugin_ui_base.hpp:90:22: error: type 'const std::array<double, 2>' does not provide a subscript operator                                                                               auto right = peak[1];                                                                                                                                                                                       ~~~~^~                                                                                                                                                                        4 errors generated.
```

**Comment by wwmm** [comment]
Well... Then something is indeed different in freebsd. Maybe is related to the use of `libc++`. But I don't see how.

**Comment by swills** [comment]
Well, the formatting there was messed up, let me try again:

```
clang++70 -Isrc/src@@pulseeffects@exe -Isrc -I../src -I../include -I. -I../ -Idata -I/usr/local/include -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include/glibmm-2.4 -I/usr/local/lib/glibmm-2.4/include -I/usr/local/include/sigc++-2.0 -I/usr/local/lib/sigc++-2.0/include -I/usr/local/include/gtk-3.0 -I/usr/local/include/pango-1.0 -I/usr/local/include/fribidi -I/usr/local/include/cairo -I/usr/local/include/pixman-1 -I/usr/local/include/freetype2 -I/usr/local/include/libdrm -I/usr/local/include/libpng16 -I/usr/local/include/harfbuzz -I/usr/local/include/gdk-pixbuf-2.0 -I/usr/local/include/gio-unix-2.0/ -I/usr/local/include/atk-1.0 -I/usr/local/include/at-spi2-atk/2.0 -I/usr/local/include/at-spi-2.0 -I/usr/local/include/dbus-1.0 -I/usr/local/lib/dbus-1.0/include -I/usr/local/include/gtkmm-3.0 -I/usr/local/lib/gtkmm-3.0/include -I/usr/local/include/atkmm-1.6 -I/usr/local/include/giomm-2.4 -I/usr/local/lib/giomm-2.4/include -I/usr/local/include/pangomm-1.4 -I/usr/local/lib/pangomm-1.4/include -I/usr/local/include/cairomm-1.0 -I/usr/local/lib/cairomm-1.0/include -I/usr/local/include/gtk-3.0/unix-print -I/usr/local/include/gdkmm-3.0 -I/usr/local/lib/gdkmm-3.0/include -I/usr/local/include/gstreamer-1.0 -I/usr/local/include/lilv-0 -I/usr/local/include/serd-0 -I/usr/local/include/sord-0 -I/usr/local/include/sratom-0 -I/usr/local/include/bs2b -Xclang -fcolor-diagnostics -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++17 -O3 '-DG_LOG_DOMAIN="pulseeffects"' -O2 -fstack-protector -isystem /usr/local/include -fno-strict-aliasing -isystem /usr/local/include -isystem /usr/local/include -pthread -D_THREAD_SAFE -D_REENTRANT  -MD -MQ 'src/src@@pulseeffects@exe/multiband_compressor_ui.cpp.o' -MF 'src/src@@pulseeffects@exe/multiband_compressor_ui.cpp.o.d' -o 'src/src@@pulseeffects@exe/multiband_compressor_ui.cpp.o' -c ../src/multiband_compressor_ui.cpp
In file included from ../src/multiband_compressor_ui.cpp:1:
In file included from ../include/multiband_compressor_ui.hpp:8:
../include/plugin_ui_base.hpp:61:21: error: type 'const std::array<double, 2>' does not provide a subscript operator
    auto left = peak[0]; 
                ~~~~^~     
../include/plugin_ui_base.hpp:62:22: error: type 'const std::array<double, 2>' does not provide a subscript operator
    auto right = peak[1]; 
                 ~~~~^~     
../include/plugin_ui_base.hpp:89:21: error: type 'const std::array<double, 2>' does not provide a subscript operator
    auto left = peak[0]; 
                ~~~~^~     
../include/plugin_ui_base.hpp:90:22: error: type 'const std::array<double, 2>' does not provide a subscript operator
    auto right = peak[1]; 
                 ~~~~^~     
4 errors generated. 
```

**Comment by swills** [comment]
> Well... Then something is indeed different in freebsd. Maybe is related to the use of `libc++`. But I don't see how.

Maybe looking at the pre-processed file on your system will show where the include is coming from there? I can post the pre-processed file from my system for comparison.

**Comment by swills** [comment]
If it's helpful, here's the preprocessed file:

https://gist.githubusercontent.com/swills/b4bf81e1f5e444a94100624b20e50097/raw/c4a4f9409a312010b62f6e0e209ba50b08b59131/null
